.github/workflows/build.yaml

```diff
@@ -28,6 +28,10 @@ jobs:
     name: build
     runs-on: ubuntu-latest
 
+    permissions:
```

Comment:
the previous pr failed https://github.com/sigstore/cosign/runs/4838213182?check_suite_focus=true

```
Run google-github-actions/auth@c6c22902f6af237edb96ede5f25a00e864589b2f
Error: google-github-actions/auth failed with: gitHub Actions did not inject $ACTIONS_ID_TOKEN_REQUEST_TOKEN or $ACTIONS_ID_TOKEN_REQUEST_URL into this job. This most likely means the GitHub Actions workflow permissions are incorrect, or this job is being run from a fork. For more information, please see https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token
```

and checking was missing to set some permissions